py/selenium/webdriver/chromium/virtual_authenticator.py

```diff
@@ -0,0 +1,248 @@
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from base64 import urlsafe_b64encode, urlsafe_b64decode
+from enum import Enum
+
+
+class Protocol(Enum):
+    """
+    Protocol to communicate with the authenticator.
+    """
+    CTAP2 = "ctap2"
+    U2F = "ctap1/u2f"
+
+
+class Transport(Enum):
+    """
+    Transport method to communicate with the authenticator.
+    """
+    BLE = "ble"
+    USB = "usb"
+    NFC = "nfc"
+    INTERNAL = "internal"
+
+
+class VirtualAuthenticatorOptions:
+
+    Protocol = Protocol
+    Transport = Transport
+
+    def __init__(self) -> None:
```

Comment:
without fully understanding the auth stuff; should these be allowed to be passed in with default values if omitted? (or some of the options)

Reply:
@symonk I'm taking the Java code as reference. Since Java is the one to first implement VirtualAuthenticator stuff and I'm blindly following that. And these are the default values passed in Java. But I'd be happy to know more learning about Authentication Stuff. For a basic Authenticator these options are fine.

- we are setting the protocol to CTAP2 which is good for resident credential and non resident credential
- we are setting the transport to USB since most auth stuff be connected to PC using that. 
- default we are making residentKey false so client will not see the key
- userConsenting should be true
- and it goes for other stuff too..